File: 0073-set-matrix-zeroes/0073-set-matrix-zeroes.java
class Solution {
    public void setZeroes(int[][] matrix) {
        int[] row = new int[matrix[0].length];
        int[] column = new int[matrix.length];
        
        for(int i=0;i<matrix.length;i++){
            for(int j=0;j<matrix[0].length;j++){
                if(matrix[i][j] == 0){
                    column[i] = 1;
                    row[j] = 1;
                }
            }
        }
        
        for(int i=0;i<matrix.length;i++){
            for(int j=0;j<matrix[0].length;j++){
                if(row[j] == 1 || column[i] == 1){
                    matrix[i][j] = 0;
                }
            }
        }
    }
}

File: 0073-set-matrix-zeroes/0073-set-matrix-zeroes.go
func setZeroes(matrix [][]int)  {
    var row = make([]int,len(matrix[0]))
    var column = make([]int,len(matrix))
    
    for i:=0;i<len(matrix);i++{
        for j:=0;j<len(matrix[0]);j++{
            if(matrix[i][j] == 0){
                row[j] = 1
                column[i] = 1 
            }
        }
    }
    
    for i:=0;i<len(matrix);i++{
        for j:=0;j<len(matrix[0]);j++{
            if(row[j] == 1 || column[i] == 1 ){
                matrix[i][j] = 0
            }
        }
    }
}